### PR TITLE
refactor: made changes to billing service and mail service for better maintainability and readability

### PR DESCRIPTION
**Pull request by iamshabell**
Close: #5985 

## PR Details

This PR refactors the `BillingService` and `MailingService` class to improve their maintainability and readability. The changes include:

- Simplifying the `reportUsage`, `setUsage`, `getMeteredEntitlement`, `getNumericEntitlement`, and `getBooleanEntitlement` methods by removing unnecessary if statements and consolidating similar code in the 
- Reformatting the code to follow a consistent code style and improve readability, including using consistent whitespace, removing unnecessary line breaks, and reducing long method signatures.
- Renaming some variables and parameters to use more descriptive names and better align with the domain concepts.
- Extracting common code blocks into separate helper functions to improve modularity and readability.

These changes should make it easier to maintain and evolve the billing service code in the future.

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by souravjain540** [comment]
@mulygottlieb can you assign someone from the team to review this PR?

**Comment by mulygottlieb** [comment]
@iamshabell I see there is no Issue connected to this PR. We do not merge PRs that are not connected to a GH issue that they fulfill/solve. Please open an issue and connect it to the PR. Keep in mind that we usually do not refactor just for the sake of refactoring if it's not connected to a task.

File: packages/amplication-server/src/core/billing/billing.service.ts [hunk 1]
```diff
@@ -229,113 +247,118 @@ export class BillingService {
...
-        BillingFeature.Services,
-        currentUsage.services
-      );
-
-      await this.setUsage(
-        workspaceId,
-        BillingFeature.ServicesAboveEntitiesPerServiceLimit,
-        currentUsage.servicesAboveEntityPerServiceLimit
-      );
+      return;
```
Comment (abrl91) [inline packages/amplication-server/src/core/billing/billing.service.ts:349]: Did you mean to:
```
if (!this.isBillingEnabled) {
  return;
}
(missing "!" in line 348)
```

File: packages/amplication-server/src/core/mail/mail.service.ts [hunk 2]
```diff
@@ -26,58 +26,60 @@ export class MailService {
...
 
       this.logger.debug("sendDeploymentNotification", args);
 
-      const msg = {
+      const message = {
         to: args.to,
-        from,
-        templateId,
+        from: fromAddress,
+        templateId: templateId,
```
Comment (abrl91) [inline packages/amplication-server/src/core/mail/mail.service.ts:72]:
```suggestion
        templateId,
```

Comment (abrl91) [inline packages/amplication-server/src/core/mail/mail.service.ts:72]:
```suggestion
        templateId,
```

**Comment by abrl91** [comment]
@iamshabell 
nicely done!
please create an issue and connect it to this PR

File: packages/amplication-server/src/core/billing/billing.service.ts (hunk 1 above)
Comment (iamshabell) [reply packages/amplication-server/src/core/billing/billing.service.ts:349]: Oh yes, fixed!

**Comment by iamshabell** [comment]
> @iamshabell nicely done! please create an issue and connect it to this PR

Connected! Thank you for the guide :)

**Review by overbit — CHANGES_REQUESTED** [review]
As we are aiming to increase our coverage, could you please add unit test to this pr to, at least cover the logic touched?

Thanks @iamshabell

**Comment by mulygottlieb** [comment]
Hi @iamshabell 
First of all, thanks a lot for your contributions to Amplication! We appreciate it a lot 💜

Having said that, I am afraid that we cannot accept this PR. As a policy, we do not refactor code without a good reason, and only when it is connected to an issue that we need to solve and is accompanied by proper test coverage.
In this case, you want this refactoring as your own initiative and then open an issue after the fact to ask for those changes, and it's not accompanied by test coverage.
While it's a seemingly straightforward change, we cannot be sure it will not surface any bugs in a code that works flawlessly today, and this is an unnecessary risk for us (hence the policy I mentioned above).

So please accept my apology; I will close this PR and the issue you've opened. You can find many options to contribute to Amplication if you wish, and you are welcome to check with our community manager @souravjain540 if he has any available issues for you to take.

Please don't be discouraged by this, and thanks again for your contribution!

**Pull request closed without merging**